server/auth/auth_oauth.py

```diff
@@ -255,8 +254,8 @@ def check_jwt_payload(self, id_token):
                 }
         if rsa_key:
             options = {}
-            if not rsa_key["n"] or not rsa_key["e"]:
-                # this is a mock auth server, do not validate
+            if self.client_id == "mock_client_id" and (not rsa_key["n"] or not rsa_key["e"]):
```

Comment:
Can you remove the testing logic from the production code and instead mock out the expected value for `options`?

Reply:
It might be a little tricky to mock out because the code here is running in a different process than the test case.

Reply:
Can you try please and let me know. It is really not a good practice to have unit testing conditionals in production code.

Reply:
One option is to have a config parameter for this.  E.g. jwt_code_options, which is a dictionary of key/values.
Another option is to switch on an environment variable.

Reply:
You are not able to `patch` the function `check_jwt_payload` and define it's return value?

Reply:
The function is run in a different process than the test.    I don't think patching will work in this case.

Reply:
I added a config parameter to handle the decode options.